Cloning poll as non-admin returns 403 and results in missing options in clone

### ⚠️ This issue respects the following points: ⚠️

- [x] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [x] This issue is **not** already reported on Github _(I've searched it)_.
- [x] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### What went wrong, what did you observe?

When attempting to clone a poll for which I do not have admin rights, the POST request to /poll/{id}/clone returns a 403 (Forbidden). Despite this, a cloned poll is still created, but the options from the original poll are missing. The error is shown as pop-up in the GUI and I can see it in the browser developer console.

### What did you expect, how polls should behave instead?

I expected that, if poll cloning is not permitted due to missing rights, the clone option is hidden or - better IMO - cloning is always possible and all options from the original would be present.

### What steps does it need to replay this bug?

1. create a poll as user A
2. share to user B
3. as user B try cloning the shared poll

### Affected polls version

8.5.0

### Installation method

Installed/updated from the appstore (Apps section of your site)

### Installation type

Updated from a much older version (more than one major version)

### Can you rule out that any extension you use is involved in the issue?

- [ ] I have checked all browser extension

### Which browser did you use, when experiencing the bug?

- [ ] Firefox
- [ ] Chrome
- [x] Chromium/Chromium based (i.e. Edge)
- [ ] Safari
- [ ] Other/Don't know

### Other browser

_No response_

### Add your browser log here

```json
relevant:1  Blocked aria-hidden on an element because its descendant retained focus. The focus must not be hidden from assistive technology users. Avoid using aria-hidden on a focused element or its ancestor. Consider using the inert attribute instead, which will also prevent focus. For more details, see the aria-hidden section of the WAI-ARIA specification at https://w3c.github.io/aria/#aria-hidden.
Element with focus: <button.action-button button-vue focusable>
Ancestor with aria-hidden: <div.v-popper__popper v-popper--theme-nc-popover-9 _ncPopover_wpltc_20 action-item__popper v-popper__popper--hidden v-popper__popper--hide-from#popper_su5c16wb_bziywt> <div id=​"popper_su5c16wb_bziywt" class=​"v-popper__popper v-popper--theme-nc-popover-9 _ncPopover_wpltc_20 action-item__popper v-popper__popper--hidden v-popper__popper--hide-to" aria-hidden=​"true" tabindex=​"0" data-popper-placement=​"bottom" style=​"position:​ absolute;​ transform:​ translate3d(1670px, 264px, 0px)​;​">​…​</div>​
PollList.vue:190   POST https://cloud.domain.de/index.php/apps/polls/poll/123/clone 403 (Forbidden)
(anonymous) @ adapters.js:69
xhr @ adapters.js:69
dd @ Axios.js:103
Promise.then
_request @ focus-trap.esm.js:76
request @ Axios.js:227
(anonymous) @ AxiosHeaders.js:253
clonePoll @ marked.esm.js:13
clone @ marked.esm.js:13
z @ index.mjs:32
q @ PollList.vue:190
C.value.s.onClick.p.<computed>.p.<computed> @ PollList.vue:190
Oo @ index.mjs:32
cu @ index.mjs:32
h1 @ index.mjs:32
onClick @ NcActionButton-BAaRMesp.mjs:224
handleClick @ NcActionButton-BAaRMesp.mjs:224
M.onClick.r.<computed>.r.<computed> @ NcActionButton-BAaRMesp.mjs:224
Oo @ index.mjs:32
cu @ index.mjs:32
u @ index.mjs:32
PollList.vue:190  [ERROR] polls: Error cloning poll {app: 'polls', uid: 'userid', level: 2, error: ye, payload: {…}}app: "polls"error: ye {message: 'Request failed with status code 403', name: 'AxiosError', code: 'ERR_BAD_REQUEST', config: {…}, request: XMLHttpRequest, …}level: 2payload: {pollId: 123}uid: "userid"[[Prototype]]: Object
log @ purify.es.mjs:1342
error @ index.js:49
clone @ marked.esm.js:13
await in clone
z @ index.mjs:32
q @ PollList.vue:190
C.value.s.onClick.p.<computed>.p.<computed> @ PollList.vue:190
Oo @ index.mjs:32
cu @ index.mjs:32
h1 @ index.mjs:32
onClick @ NcActionButton-BAaRMesp.mjs:224
handleClick @ NcActionButton-BAaRMesp.mjs:224
M.onClick.r.<computed>.r.<computed> @ NcActionButton-BAaRMesp.mjs:224
Oo @ index.mjs:32
cu @ index.mjs:32
u @ index.mjs:32














relevant:1  Blocked aria-hidden on an element because its descendant retained focus. The focus must not be hidden from assistive technology users. Avoid using aria-hidden on a focused element or its ancestor. Consider using the inert attribute instead, which will also prevent focus. For more details, see the aria-hidden section of the WAI-ARIA specification at https://w3c.github.io/aria/#aria-hidden.
Element with focus: <button.action-button button-vue focusable>
Ancestor with aria-hidden: <div.v-popper__popper v-popper--theme-nc-popover-9 _ncPopover_wpltc_20 action-item__popper v-popper__popper--hidden v-popper__popper--hide-from#popper_su5c16wb_bziywt> <div id=​"popper_su5c16wb_bziywt" class=​"v-popper__popper v-popper--theme-nc-popover-9 _ncPopover_wpltc_20 action-item__popper v-popper__popper--hidden v-popper__popper--hide-to" aria-hidden=​"true" tabindex=​"0" data-popper-placement=​"bottom" style=​"position:​ absolute;​ transform:​ translate3d(1670px, 264px, 0px)​;​">​…​</div>​
PollList.vue:190   POST https://cloud.domain.de/index.php/apps/polls/poll/123/clone 403 (Forbidden)
(anonymous) @ adapters.js:69
xhr @ adapters.js:69
dd @ Axios.js:103
Promise.then
_request @ focus-trap.esm.js:76
request @ Axios.js:227
(anonymous) @ AxiosHeaders.js:253
clonePoll @ marked.esm.js:13
clone @ marked.esm.js:13
z @ index.mjs:32
q @ PollList.vue:190
C.value.s.onClick.p.<computed>.p.<computed> @ PollList.vue:190
Oo @ index.mjs:32
cu @ index.mjs:32
h1 @ index.mjs:32
onClick @ NcActionButton-BAaRMesp.mjs:224
handleClick @ NcActionButton-BAaRMesp.mjs:224
M.onClick.r.<computed>.r.<computed> @ NcActionButton-BAaRMesp.mjs:224
Oo @ index.mjs:32
cu @ index.mjs:32
u @ index.mjs:32
PollList.vue:190  [ERROR] polls: Error cloning poll {app: 'polls', uid: 'userid', level: 2, error: ye, payload: {…}}app: "polls"error: ye {message: 'Request failed with status code 403', name: 'AxiosError', code: 'ERR_BAD_REQUEST', config: {…}, request: XMLHttpRequest, …}level: 2payload: {pollId: 123}uid: "userid"[[Prototype]]: Object
log @ purify.es.mjs:1342
error @ index.js:49
clone @ marked.esm.js:13
await in clone
z @ index.mjs:32
q @ PollList.vue:190
C.value.s.onClick.p.<computed>.p.<computed> @ PollList.vue:190
Oo @ index.mjs:32
cu @ index.mjs:32
h1 @ index.mjs:32
onClick @ NcActionButton-BAaRMesp.mjs:224
handleClick @ NcActionButton-BAaRMesp.mjs:224
M.onClick.r.<computed>.r.<computed> @ NcActionButton-BAaRMesp.mjs:224
Oo @ index.mjs:32
cu @ index.mjs:32
u @ index.mjs:32









{
    "app": "polls",
    "uid": "userid",
    "level": 2,
    "error": {
        "message": "Request failed with status code 403",
        "name": "AxiosError",
        "stack": "AxiosError: Request failed with status code 403\n    at Jp (https://cloud.domain.de/apps/polls/js/_plugin-vue_export-helper-DWCfimU6.chunk.mjs:8:1083)\n    at XMLHttpRequest.k (https://cloud.domain.de/apps/polls/js/_plugin-vue_export-helper-DWCfimU6.chunk.mjs:8:5704)\n    at Vn.request (https://cloud.domain.de/apps/polls/js/_plugin-vue_export-helper-DWCfimU6.chunk.mjs:10:2071)\n    at async Proxy.clone (https://cloud.domain.de/apps/polls/js/session-Ct8iNBZc.chunk.mjs:2:45956)\n    at async q (https://cloud.domain.de/apps/polls/js/PollList-C5eFsbPw.chunk.mjs:2:18195)",
        "config": {
            "transitional": {
                "silentJSONParsing": true,
                "forcedJSONParsing": true,
                "clarifyTimeoutError": false
            },
            "adapter": [
                "xhr",
                "http",
                "fetch"
            ],
            "transformRequest": [
                null
            ],
            "transformResponse": [
                null
            ],
            "timeout": 0,
            "xsrfCookieName": "XSRF-TOKEN",
            "xsrfHeaderName": "X-XSRF-TOKEN",
            "maxContentLength": -1,
            "maxBodyLength": -1,
            "env": {},
            "headers": {
                "Accept": "application/json",
                "Content-Type": "application/x-www-form-urlencoded",
                "requesttoken": "JsGDoEtWvCf7Ws+4ujyGaBbKvylbFTWlFvbf8EACPoA=:EojX6T0dyhScALnS1H7xHn29inghIB7rJLiqnSV0V8E=",
                "X-Requested-With": "XMLHttpRequest",
                "Nc-Polls-Client-Time-Zone": "Europe/Berlin",
                "Nc-Polls-Client-Id": "0pyv93mv5y7f"
            },
            "baseURL": "/index.php/apps/polls/",
            "method": "post",
            "url": "poll/123/clone",
            "cancelToken": {
                "promise": {},
                "_listeners": null,
                "reason": {
                    "message": "clonePoll canceled",
                    "name": "CanceledError",
                    "stack": "CanceledError: clonePoll canceled\n    at Object.cancel (https://cloud.domain.de/apps/polls/js/_plugin-vue_export-helper-DWCfimU6.chunk.mjs:11:2533)\n    at Object.handleRequestCancellation (https://cloud.domain.de/apps/polls/js/session-Ct8iNBZc.chunk.mjs:2:953)\n    at Object.clonePoll (https://cloud.domain.de/apps/polls/js/session-Ct8iNBZc.chunk.mjs:2:7143)\n    at Proxy.clone (https://cloud.domain.de/apps/polls/js/session-Ct8iNBZc.chunk.mjs:2:45964)\n    at Proxy.z (https://cloud.domain.de/apps/polls/js/_plugin-vue_export-helper-DWCfimU6.chunk.mjs:2:111894)\n    at q (https://cloud.domain.de/apps/polls/js/PollList-C5eFsbPw.chunk.mjs:2:18203)\n    at C.value.s.onClick.p.<computed>.p.<computed> (https://cloud.domain.de/apps/polls/js/PollList-C5eFsbPw.chunk.mjs:2:18807)\n    at Oo (https://cloud.domain.de/apps/polls/js/_plugin-vue_export-helper-DWCfimU6.chunk.mjs:2:20497)\n    at cu (https://cloud.domain.de/apps/polls/js/_plugin-vue_export-helper-DWCfimU6.chunk.mjs:2:20568)\n    at h1 (https://cloud.domain.de/apps/polls/js/_plugin-vue_export-helper-DWCfimU6.chunk.mjs:2:57613)",
                    "code": "ERR_CANCELED"
                }
            },
            "allowAbsoluteUrls": true
        },
        "code": "ERR_BAD_REQUEST",
        "status": 403
    },
    "payload": {
        "pollId": 123
    }
}
```

### Additional client environment information

_No response_

### NC version

Nextcloud 27

### Other Nextcloud version

_No response_

### PHP engine version

PHP 8.1

### Other PHP version

_No response_

### Database engine

MySQL

### Database Engine version or other Database

_No response_

### Which user-backends are you using?

- [x] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other/Don't know

### Add your nextcloud server log here

```json

```

### Additional environment informations

_No response_

### Configuration report

```shell

```

### List of activated Apps

```shell

```

### Nextcloud Signing status

```shell

```

### Additional Information

As always, I very much appreciate the work that goes into this app!

Cloning as share-receiver with admin rights works fine.

I've recently started digging into NC app development myself, so if there is a prefered way on how to address this, I'm open to try & submit a PR for this :)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@ All notable changes to this project will be documented in this file.
 ## [unreleased]
 ### Fixed
  - Changed sql query to be ansi safe
+ - Cloning polls
 
 ### Added
  - Show locale date as tooltip for relative date time information

--- a/lib/Service/OptionService.php
+++ b/lib/Service/OptionService.php
@@ -332,7 +332,9 @@ class OptionService {
 	 */
 	public function clone(int $fromPollId, int $toPollId): void {
 		$this->pollMapper->get($fromPollId)
-			->request(Poll::PERMISSION_POLL_ACCESS)
+			->request(Poll::PERMISSION_POLL_ACCESS);
+
+		$this->pollMapper->get($toPollId)
 			->request(Poll::PERMISSION_OPTION_ADD);
 
 		foreach ($this->optionMapper->findByPoll($fromPollId) as $origin) {

